parsl/dataflow/strategy.py

```diff
@@ -275,3 +277,132 @@ def _strategy_simple(self, status_list, tasks, *args, kind=None, **kwargs):
             else:
                 # logger.debug("Strategy: Case 3")
                 pass
+
+    def _strategy_htex(self, tasks, *args, kind=None, **kwargs):
+        """Peek at the DFK and the executors specified.
+
+        We assume here that tasks are not held in a runnable
+        state, and that all tasks from an app would be sent to
+        a single specific executor, i.e tasks cannot be specified
+        to go to one of more executors.
+
+        Args:
+            - tasks (task_ids): Not used here.
+
+        KWargs:
+            - kind (Not used)
+        """
+
+        for label, executor in self.dfk.executors.items():
+            if not executor.scaling_enabled:
+                continue
+
+            # Tasks that are either pending completion
+            active_tasks = executor.outstanding
+
+            status = executor.status()
+            self.unset_logging()
+
+            # FIXME we need to handle case where provider does not define these
+            # FIXME probably more of this logic should be moved to the provider
+            min_blocks = executor.provider.min_blocks
+            max_blocks = executor.provider.max_blocks
+            if isinstance(executor, HighThroughputExecutor):
+
+                tasks_per_node = executor.workers_per_node
+            elif isinstance(executor, ExtremeScaleExecutor):
```

Comment:
this if will never fire, because an ExtremeScaleExcecutor is a HighThroughputExecutor and will be captured by the previous case.

Reply:
I think we are better off raising an exception if the executor isn't of type HTEX. What do you think ?

Reply:
I don't have a clear idea of what to do when a strategy goes wrong... strategy exceptions at the moment propagate upwards into flow control and raise this:
```
    logger.error("Flow control callback threw an exception - logging and proceeding anyway", exc_info=True)
```

It bugs me that this doesn't work for WQ, which also has providers and scale_out/in behaviour. That's probably something for another time, though.